subprojects/common/report-viewer/src/main/kotlin/com/avito/report/TestArtifactsProviderFactory.kt

```diff
@@ -0,0 +1,46 @@
+package com.avito.report
+
+import com.avito.report.internal.DirectTestArtifactsProvider
+import com.avito.report.internal.UniqueDirTestArtifactsProvider
+import com.avito.report.model.TestStaticData
+import java.io.File
+
+object TestArtifactsProviderFactory {
+
+    fun create(
+        testReportRootDir: Lazy<File>,
+        testStaticData: TestStaticData
+    ): TestArtifactsProvider {
+        return UniqueDirTestArtifactsProvider(
+            rootDir = testReportRootDir,
+            testDirGenerator = TestDirGenerator.StaticData(testStaticData)
+        )
+    }
+
+    fun create(
+        testReportRootDir: Lazy<File>,
+        className: String,
+        methodName: String
+    ): TestArtifactsProvider {
+        return UniqueDirTestArtifactsProvider(
+            rootDir = testReportRootDir,
+            testDirGenerator = TestDirGenerator.Impl(className, methodName)
+        )
+    }
+
+    fun createForTempDir(tempDir: File): TestArtifactsProvider {
```

Comment:
Can and should we know about "temp" nature of the dir at this level?
Maybe we could express other attributes of dir here if they are important.